arch/arc/core/mpu/arc_mpu_v2_internal.h

```diff
@@ -118,7 +118,7 @@ static inline bool _is_in_region(u32_t r_index, u32_t start, u32_t size)
 	r_size_lshift = (r_size_lshift & 0x3) | ((r_size_lshift >> 7) & 0x1C);
 	r_addr_end = r_addr_start  + (1 << (r_size_lshift + 1));
 
-	if (start >= r_addr_start && (start + size) < r_addr_end) {
+	if (start >= r_addr_start && (start + size) <= r_addr_end) {
```

Comment:
Don't know what the best is, here. I would prefer : `(start + size - 1) < r_addr_end`
It's a style thing.

Reply:
you've lost me.
they are the same.

Reply:
I know, I said it is a style thing :) We usually have "something" -1 < "sthg else", not <=.
Just style, ignore at will.

Reply:
my suggestion would be to directly compare sizes and not calculate end addresses: no need to add the start addresses, less risk for 1-off mistakes.